Index: Python-2.5.5/Include/Python.h
===================================================================
--- Python-2.5.5.orig/Include/Python.h	2010-04-26 23:36:53.838663021 +0100
+++ Python-2.5.5/Include/Python.h	2010-04-26 23:37:09.855664303 +0100
@@ -162,7 +162,7 @@
 #endif
 
 /* Define macros for inline documentation. */
-#define PyDoc_VAR(name) static char name[]
+#define PyDoc_VAR(name) static const char name[]
 #define PyDoc_STRVAR(name,str) PyDoc_VAR(name) = PyDoc_STR(str)
 #ifdef WITH_DOC_STRINGS
 #define PyDoc_STR(str) str
