Index: Python-2.5.5/Include/token.h
===================================================================
--- Python-2.5.5.orig/Include/token.h	2010-04-26 23:36:53.610647709 +0100
+++ Python-2.5.5/Include/token.h	2010-04-26 23:37:10.631710383 +0100
@@ -72,7 +72,7 @@
 #define ISEOF(x)		((x) == ENDMARKER)
 
 
-PyAPI_DATA(char *) _PyParser_TokenNames[]; /* Token names */
+PyAPI_DATA(const char * const) _PyParser_TokenNames[]; /* Token names */
 PyAPI_FUNC(int) PyToken_OneChar(int);
 PyAPI_FUNC(int) PyToken_TwoChars(int, int);
 PyAPI_FUNC(int) PyToken_ThreeChars(int, int, int);
Index: Python-2.5.5/Parser/tokenizer.c
===================================================================
--- Python-2.5.5.orig/Parser/tokenizer.c	2010-04-26 23:36:53.626649187 +0100
+++ Python-2.5.5/Parser/tokenizer.c	2010-04-26 23:37:10.644196721 +0100
@@ -41,7 +41,7 @@
 
 /* Token names */
 
-char *_PyParser_TokenNames[] = {
+const char * const _PyParser_TokenNames[] = {
 	"ENDMARKER",
 	"NAME",
 	"NUMBER",
Index: Python-2.5.5/Include/grammar.h
===================================================================
--- Python-2.5.5.orig/Include/grammar.h	2010-04-26 23:36:53.594648289 +0100
+++ Python-2.5.5/Include/grammar.h	2010-04-26 23:37:10.666385139 +0100
@@ -76,7 +76,7 @@
 
 int addlabel(labellist *ll, int type, char *str);
 int findlabel(labellist *ll, int type, char *str);
-char *PyGrammar_LabelRepr(label *lb);
+const char *PyGrammar_LabelRepr(label *lb);
 void translatelabels(grammar *g);
 
 void addfirstsets(grammar *g);
Index: Python-2.5.5/Parser/grammar1.c
===================================================================
--- Python-2.5.5.orig/Parser/grammar1.c	2010-04-26 23:36:53.646651177 +0100
+++ Python-2.5.5/Parser/grammar1.c	2010-04-26 23:37:10.666385139 +0100
@@ -30,7 +30,7 @@
 #endif
 }
 
-char *
+const char *
 PyGrammar_LabelRepr(label *lb)
 {
 	static char buf[100];
