Index: Python-2.5.2/Modules/Setup.dist
===================================================================
--- Python-2.5.2.orig/Modules/Setup.dist	2008-08-20 18:09:20.000000000 +0100
+++ Python-2.5.2/Modules/Setup.dist	2008-08-20 18:09:58.000000000 +0100
@@ -109,16 +109,16 @@
 # This only contains the minimal set of modules required to run the 
 # setup.py script in the root of the Python source tree.
 
-posix posixmodule.c		# posix (UNIX) system calls
-errno errnomodule.c		# posix (UNIX) errno values
-pwd pwdmodule.c			# this is needed to find out the user's home dir
+#posix posixmodule.c		# posix (UNIX) system calls
+#errno errnomodule.c		# posix (UNIX) errno values
+#pwd pwdmodule.c			# this is needed to find out the user's home dir
 				# if $HOME is not set
-_sre _sre.c			# Fredrik Lundh's new regular expressions
-_codecs _codecsmodule.c		# access to the builtin codecs and codec registry
+#_sre _sre.c			# Fredrik Lundh's new regular expressions
+#_codecs _codecsmodule.c		# access to the builtin codecs and codec registry
 
 # The zipimport module is always imported at startup. Having it as a
 # builtin module avoids some bootstrapping problems and reduces overhead.
-zipimport zipimport.c
+#zipimport zipimport.c
 
 # The rest of the modules listed in this file are all commented out by
 # default.  Usually they can be detected and built as dynamically
@@ -131,7 +131,7 @@
 # ======================================================================
 
 # The Python symtable module depends on .h files that setup.py doesn't track
-_symtable symtablemodule.c
+#_symtable symtablemodule.c
 
 # The SGI specific GL module:
 
@@ -484,4 +484,4 @@
 # xx xxmodule.c
 
 # Another example -- the 'xxsubtype' module shows C-level subtyping in action
-xxsubtype xxsubtype.c
+#xxsubtype xxsubtype.c
