Index: quilted/Include/Python.h
===================================================================
--- quilted.orig/Include/Python.h	2006-06-10 13:23:46.000000000 +0100
+++ quilted/Include/Python.h	2008-09-04 15:57:24.000000000 +0100
@@ -162,7 +162,7 @@
 #endif
 
 /* Define macros for inline documentation. */
-#define PyDoc_VAR(name) static char name[]
+#define PyDoc_VAR(name) static const char name[]
 #define PyDoc_STRVAR(name,str) PyDoc_VAR(name) = PyDoc_STR(str)
 #ifdef WITH_DOC_STRINGS
 #define PyDoc_STR(str) str
