Index: quilted/Include/token.h
===================================================================
--- quilted.orig/Include/token.h	2008-09-04 16:09:21.000000000 +0100
+++ quilted/Include/token.h	2008-09-04 16:09:30.000000000 +0100
@@ -72,7 +72,7 @@
 #define ISEOF(x)		((x) == ENDMARKER)
 
 
-PyAPI_DATA(char *) _PyParser_TokenNames[]; /* Token names */
+PyAPI_DATA(const char *) _PyParser_TokenNames[]; /* Token names */
 PyAPI_FUNC(int) PyToken_OneChar(int);
 PyAPI_FUNC(int) PyToken_TwoChars(int, int);
 PyAPI_FUNC(int) PyToken_ThreeChars(int, int, int);
Index: quilted/Parser/tokenizer.c
===================================================================
--- quilted.orig/Parser/tokenizer.c	2008-09-04 16:08:59.000000000 +0100
+++ quilted/Parser/tokenizer.c	2008-09-04 16:09:09.000000000 +0100
@@ -41,7 +41,7 @@
 
 /* Token names */
 
-char *_PyParser_TokenNames[] = {
+const char *_PyParser_TokenNames[] = {
 	"ENDMARKER",
 	"NAME",
 	"NUMBER",
Index: quilted/Include/grammar.h
===================================================================
--- quilted.orig/Include/grammar.h	2008-09-04 16:18:31.000000000 +0100
+++ quilted/Include/grammar.h	2008-09-04 16:18:57.000000000 +0100
@@ -76,7 +76,7 @@
 
 int addlabel(labellist *ll, int type, char *str);
 int findlabel(labellist *ll, int type, char *str);
-char *PyGrammar_LabelRepr(label *lb);
+const char *PyGrammar_LabelRepr(label *lb);
 void translatelabels(grammar *g);
 
 void addfirstsets(grammar *g);
Index: quilted/Parser/grammar1.c
===================================================================
--- quilted.orig/Parser/grammar1.c	2008-09-04 16:18:02.000000000 +0100
+++ quilted/Parser/grammar1.c	2008-09-04 16:18:19.000000000 +0100
@@ -30,7 +30,7 @@
 #endif
 }
 
-char *
+const char *
 PyGrammar_LabelRepr(label *lb)
 {
 	static char buf[100];
