Index: Python-2.5.5/Python/getcopyright.c
===================================================================
--- Python-2.5.5.orig/Python/getcopyright.c	2010-04-26 23:36:53.898667719 +0100
+++ Python-2.5.5/Python/getcopyright.c	2010-04-26 23:37:09.491640952 +0100
@@ -2,7 +2,7 @@
 
 #include "Python.h"
 
-static char cprt[] = 
+static const char cprt[] = 
 "\
 Copyright (c) 2001-2008 Python Software Foundation.\n\
 All Rights Reserved.\n\
