Index: Python-2.5.5/Objects/longobject.c
===================================================================
--- Python-2.5.5.orig/Objects/longobject.c	2010-04-26 23:36:53.782658697 +0100
+++ Python-2.5.5/Objects/longobject.c	2010-04-26 23:37:10.219687121 +0100
@@ -1342,7 +1342,7 @@
  * Note that when converting a base B string, a char c is a legitimate
  * base B digit iff _PyLong_DigitValue[Py_CHARMASK(c)] < B.
  */
-int _PyLong_DigitValue[256] = {
+const int _PyLong_DigitValue[256] = {
 	37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
 	37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
 	37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37, 37,
Index: Python-2.5.5/Include/longobject.h
===================================================================
--- Python-2.5.5.orig/Include/longobject.h	2010-04-26 23:36:53.762658223 +0100
+++ Python-2.5.5/Include/longobject.h	2010-04-26 23:37:10.227683490 +0100
@@ -25,7 +25,7 @@
 PyAPI_FUNC(Py_ssize_t) _PyLong_AsSsize_t(PyObject *);
 PyAPI_FUNC(PyObject *) _PyLong_FromSize_t(size_t);
 PyAPI_FUNC(PyObject *) _PyLong_FromSsize_t(Py_ssize_t);
-PyAPI_DATA(int) _PyLong_DigitValue[256];
+PyAPI_DATA(const int) _PyLong_DigitValue[256];
 
 /* _PyLong_AsScaledDouble returns a double x and an exponent e such that
    the true value is approximately equal to x * 2**(SHIFT*e).  e is >= 0.
